File: Algorithms/Java/Taller02/Genome.java
public class Genome{
    public int SubseqLength(String Seq1, String Seq2){
        return pairStar(Seq1, Seq2, "", 0).length();
    }
    
    public String longSubseq(String str1, String str2){
        return pairStar(str1, str2, "", 0);
    }
    
    private String pairStar(String str1, String str2, String answer, int pivote) {
        int may = (str1.length() > str2.length()) ? str1.length() : str2.length();
        int min = (may == str1.length()) ? str2.length() : str1.length();
  
        if(min != 0 && pivote < str2.length()){
            if(str1.charAt(0) == str2.charAt(pivote))
            return pairStar(str1.substring(1), this.reWrite(str2, pivote), answer + str1.charAt(0), 0);
            return pairStar(str1, str2, answer, pivote + 1);
        } if(min == 0) return answer;
        return pairStar(str1.substring(1), str2, answer, 0);
    }
    
    private String reWrite(String write, int Elim){
        if(Elim != 0 && Elim < write.length()-1)
        return write.substring(0,Elim) + write.substring(Elim+1);
        else if(Elim == 0)
        return write.substring(1);
        else if(Elim == write.length()-1)
        return write.substring(0,Elim);
        return write;
    }
}

File: Algorithms/Python/Taller02/Genome.py
class Genome():

    def SubseqLength(self, Sequence1, Sequence2):
        return len(self.CompareStrings(Sequence1, Sequence2, "", 0))

    def LongSubseq(self, Sequence1, Sequence2):
        return self.CompareStrings(Sequence1, Sequence2, "", 0)

    def CompareStrings(self, Sqq1, Sqq2, Answer, Swivel):
        if len(Sqq1) == 0 or len(Sqq2) == 0: return Answer
        else:
            if Swivel < len(Sqq2):
                if Sqq1[0] == Sqq2[Swivel]: return self.CompareStrings(Sqq1[1:], self.Rewrite(Sqq2, Swivel), Answer + Sqq1[0], 0)
                else: return self.CompareStrings(Sqq1, Sqq2, Answer, Swivel + 1)
        return self.CompareStrings(Sqq1[1:], Sqq2, Answer, 0)

    def Rewrite(self, Write, Index):
        if Index != 0 and Index < len(Write) - 1: return Write[0:Index] + Write[Index + 1:]
        elif Index == 0: return Write[1:]
        elif Index == len(Write) - 1: return Write[0:Index]
        return Write

def main():
    Aux = Genome()
    Sqq1 = input("First Sequence: ")
    Sqq2 = input("Second Sequence: ")
    print(Aux.LongSubseq(Sqq1, Sqq2))
    print("Common Sequence's Length:", Aux.SubseqLength(Sqq1, Sqq2))

if __name__ == "__main__":
    main()
